java/java-impl-inspections/src/com/intellij/codeInspection/CastCanBeReplacedWithVariableInspection.java

```diff
@@ -116,7 +113,52 @@ private static PsiVariable findReplacement(@NotNull PsiMethod method,
       }
     }
 
-    return null;
+    List<PsiAssignmentExpression> narrowTypeFound =
+      SyntaxTraverser.psiTraverser(method)
+        .filter(PsiAssignmentExpression.class)
+        .filter(assignment -> EquivalenceChecker.getCanonicalPsiEquivalence().expressionsAreEquivalent(assignment.getLExpression(), operand))
+        .filter(assignment -> {
+          PsiExpression typeElement = assignment.getRExpression();
+          return typeElement != null && InstanceOfUtils.typeCompatible(typeElement.getType(), castType, operand);
```

Comment:
Add type check for `null`, please. Otherwise, the exception 
`java.lang.IllegalArgumentException: java.lang.IllegalArgumentException: Argument for @NotNull parameter 'instanceOfType' of com/siyeh/ig/psiutils/InstanceOfUtils.typeCompatible must not be null`
will be thrown for this code:
```
void method() {
  String anActualText = "Hello World!";
  Object aText;
  aText = unresolvedMethodCall(anActualText);
  System.out.println(((String) aText).trim());
}
```

And also for line 144.

Reply:
Done